src/app/components/shared/typeahead-input/typeahead-input.component.ts

```diff
@@ -0,0 +1,112 @@
+import { Component, ElementRef, Input, ViewChild } from "@angular/core";
+import { AbstractModel } from "@models/AbstractModel";
+import { NgbTypeaheadSelectItemEvent } from "@ng-bootstrap/ng-bootstrap";
+import {
+  BehaviorSubject,
+  debounceTime,
+  distinctUntilChanged,
+  map,
+  Observable,
+  OperatorFunction,
+  switchMap,
+} from "rxjs";
+import { defaultDebounceTime } from "src/app/app.helper";
+
+@Component({
+  selector: "baw-typeahead-input[optionsCallback]",
+  templateUrl: "typeahead-input.component.html",
+  styleUrls: ["typeahead-input.component.scss"],
+})
+export class TypeaheadInputComponent {
+  public constructor() {}
+
```

Comment:
~you need a unique key function - and you can use this to deduplicate results~

You need some way to remove duplicate results. Unique key function would work but suffers from a logic exhaustion bug.

instead we should provide a list of already selected results to the search function.

Reply:
I've added `activeItems` to the search callback, and added a condition to api requests to not include models with certain attribute values using `not_in`